__tests__/integration.test.js

```diff
@@ -458,3 +458,145 @@ describe("DELETE /api/comments/:comment_id", () => {
+      });
+  });
+  it("should sort by a given query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=title")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("title", { descending: true });
+      });
+  });
+  it("should sort by a given query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=article_id")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("article_id", { descending: true });
+      });
+  });
+  it("should sort by a given query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=topic")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("topic", { descending: true });
+      });
+  });
+  it("should sort by a given query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=votes")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("votes", { descending: true });
+      });
+  });
+  it("should sort by a given query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("created_at", {
+          coerce: true,
+          descending: true,
+        });
+      });
+  });
+  it("should 400  for an invalid sort_by query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=dodgy")
+      .expect("Content-Type", /json/)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toEqual(
+          `dodgy is not a valid argument, use ['author','title','article_id','topic','votes','created_at']`
+        );
+      });
+  });
+  it("should order by a given query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at&order=ASC")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("created_at", {
+          coerce: true,
+          ascending: true,
+        });
+      });
+  });
+  it("should 400 for an invalid order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at&order=dodgy")
+      .expect("Content-Type", /json/)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("DODGY is not a valid order use [ASC, DESC]");
+      });
+  });
+});
```

Comment:
I think there is one test missing here, a 200 for a completely valid sort_by with no articles :)